torch/distributed/_object.py

```diff
@@ -0,0 +1,53 @@
+import io
+import torch
+
+
+def _object_to_tensor(obj):
+    """
+    Serialize Python object to torch.ByteTensor.
+    """
+    buf = io.BytesIO()
+    torch.save(obj, buf)
+    return torch.ByteTensor(list(buf.getvalue()))
+
+
+def _tensor_to_object(tensor):
+    """
+    Deserialize Python object from torch.ByteTensor.
+    """
+    buf = io.BytesIO(bytearray(tensor.tolist()))
```

Comment:
Ugh, `tolist` is likely to be awfully slow. Can't we e.g. convert to numpy (if available), and use the buffer interface (e.g. `memoryview`) to initialize the IO buffer?